src/Listener/RelatedModelsListener.php

```diff
@@ -51,6 +52,27 @@ public function beforePaginate(Event $event)
     }
 
     /**
+     * Automatically parse and contain related table classes
+     *
+     * @param \Cake\Event\Event $event Before find event
+     * @return void
+     */
+    public function beforeFind(Event $event)
+    {
+        $contained = $event->subject()->query->contain();
```

Comment:
Or make a put the common code in a protected function `_eventHandler()` and call that from both the callback methods.